### PR TITLE
httpput-postfields.c: new example doing PUT with POSTFIELDS

### PR DESCRIPTION
**Pull request by bagder**
Proposed-by: Jeroen Ooms
Ref: #6186

/cc @jeroen

### CODE REVIEW

**Comment by jay** [comment]
Is this common? I think we'd want to discourage use of customrequest.

**Comment by bagder** [comment]
Yes we do. Using the approach shown here won't work if there's a redirect since `FOLLOWLOCATION` and `CUSTOMREQUEST` doesn't really go well together...

**Comment by jeroen** [comment]
@jay Many REST apis require you to do a PUT with a little json/text payload. For example this is what I am working on now, it has many such examples: https://docs.github.com/en/free-pro-team@latest/rest/overview/endpoints-available-for-github-apps

Using PUT to perform a file-upload is actually very rare in practice, I think most users are more interested in the former.

File: docs/examples/httpput-postfields.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+  url = argv[1];
+
+  /* In windows, this will init the winsock stuff */
+  curl_global_init(CURL_GLOBAL_ALL);
+
+  /* get a curl handle */
+  curl = curl_easy_init();
+  if(curl) {
+    struct curl_slist *headers = NULL;
```
Comment (emilengler) [inline docs/examples/httpput-postfields.c:63]: I wouldn't use NULL unless stddef.h is included

**Comment by bagder** [comment]
@jeroen The point isn't really what the PUT is used for, it is rather the inherent problems with using customrequest. Primarily the fact that it often breaks redirect-following. I will add a comment to the example addressing that.

File: docs/examples/httpput-postfields.c (hunk 1 above)
Comment (bagder) [reply docs/examples/httpput-postfields.c:63]: We already use NULL in numerous examples and I don't think any of them include stddef.h... And they build fine on many platforms.

**Pull request closed without merging**